elasticgraph-schema_definition/lib/elastic_graph/schema_definition/factory.rb

```diff
@@ -154,6 +156,18 @@ def new_filter_input_type(source_type, name_prefix: source_type, category: :filt
             EOS
           end
 
+          t.field @state.schema_elements.all_of, "[#{t.name}!]" do |f|
+            f.documentation <<~EOS
+              Matches records where all of the provided sub-filters evaluate to true. This works just like an AND operator in SQL.
+
+              Note: multiple filters are automatically ANDed together. This is only needed when you have multiple filters that can't
+              be provided on a single `#{t.name}` input because of collisions between key names. For example, if you want to provide
+              multiple `#{any_satisfy}: ...` filters, you could do `#{all_of}: [{#{any_satisfy}: ...}, {#{any_satisfy}: ...}]`.
```

Comment:
`any_satisfy` is only available on list filters. It was a great example for when we only offered `all_of` on a list filter, but may be confusing to users now that `all_of` shows up everywhere.  How about this instead:

> For example, if you want to AND multiple OR'd sub-filters (the equivalent of `(A OR B) AND (C OR D)`),
> you could do `#{all_of}: [{#{any_of}: [...]}, {#{any_of}: [...]}]`.